substratevm/src/com.oracle.svm.jfr/src/com/oracle/svm/jfr/JfrSymbolRepository.java

```diff
@@ -110,14 +110,22 @@ public void write(JfrChunkWriter writer) throws IOException {
 
     private void writeSymbol(JfrChunkWriter writer, JfrSymbol symbol) throws IOException {
         writer.writeCompressedLong(symbol.getId());
-        byte utf8 = 3;
-        writer.writeByte(utf8);
-        writer.writeCompressedInt(symbol.getValue().length());
-        String value = symbol.getValue();
+        writer.writeByte(JfrChunkWriter.TYPE_UTF8);
+        byte[] value = symbol.getValue().getBytes(StandardCharsets.UTF_8);
         if (symbol.getReplaceDotWithSlash()) {
-            value = value.replaceAll("\\.", "\\");
+            replaceDotWithSlash(value);
+        }
+        writer.writeCompressedInt(value.length);
+        writer.writeBytes(value);
+    }
+
+    private void replaceDotWithSlash(byte[] utf8String) {
+        for (int i = 0; i < utf8String.length; i++) {
+            // '.' is 46 and '/' is 47 for StandardCharsets.UTF_8
+            if (utf8String[i] == 46) {
```

Comment:
It should be possible to compare against the characters '.' and '/'. If that does not work for some reason, please define named constants instead.